go/kbfs/libkbfs/subscription_manager_test.go

```diff
@@ -0,0 +1,91 @@
+package libkbfs
+
+import (
+	"testing"
+	"time"
+
+	gomock "github.com/golang/mock/gomock"
+	"github.com/keybase/client/go/kbfs/data"
+	"github.com/keybase/client/go/kbfs/favorites"
+	"github.com/keybase/client/go/kbfs/tlf"
+	keybase1 "github.com/keybase/client/go/protocol/keybase1"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/net/context"
+)
+
+func delayedFinish(ctl *gomock.Controller, timeout time.Duration) (
+	finish func(), done func(args ...interface{})) {
+	ch := make(chan struct{})
+	return func() {
+			select {
+			case <-time.After(timeout):
+			case <-ch:
+			}
+			ctl.Finish()
+		}, func(args ...interface{}) {
+			ch <- struct{}{}
+		}
+}
+
+func initSubscriptionMagagerTest(t *testing.T) (config Config,
+	subscriber Subscriber, notifier *MockSubscriptionNotifier,
+	finish func(), lastMockDone func(args ...interface{})) {
+	ctl := gomock.NewController(t)
+	finish, lastMockDone = delayedFinish(ctl, time.Second)
+	config = MakeTestConfigOrBust(t, "jdoe")
+	notifier = NewMockSubscriptionNotifier(ctl)
+	subscriber = config.SubscriptionManager().Subscriber(notifier)
+	return config, subscriber, notifier, finish, lastMockDone
+}
+
+func TestSubscriptionManagerSubscribePath(t *testing.T) {
```

Comment:
It would be great to include file writes (before the `SyncAll` call) in this test, since they use a different notification path (`LocalChange`).